src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/finalclass/InputFinalClass.java

```diff
@@ -127,3 +129,52 @@ private SomeClass() {}
         private SomeClass() {}
     }
 }
+
+class test8 {
+    public static final test8 ONE = new test8() {
+        @Override
+        public int value() {
+            return 1;
+        }
+    };
+
+    private test8() {
+    }
+
+    public int value() {
+        return 0;
+    }
+}
+
+class test {
+
+    private test(String s) {
+        String a = "hello" + s;
+    }
+
+    public int count() {
+        return 1;
+    }
+    public static final test ONE = new test("world");
+    public static final test6 TWO = new test6() {
+    };
+
+    public static void main(String[] args) {
```

Comment:
is this code useful for test input? Please remove if not

Reply:
Yes it is, needed for coverage with `LITERAL_NEW`